homeassistant/components/rfxtrx/sensor.py

```diff
@@ -133,16 +290,16 @@ def sensor_update(event, device_id):
 class RfxtrxSensor(RfxtrxEntity, SensorEntity):
     """Representation of a RFXtrx sensor."""
 
-    def __init__(self, device, device_id, data_type, event=None):
+    entity_description: RfxtrxSensorEntityDescription
+
+    def __init__(self, device, device_id, entity_description, event=None):
         """Initialize the sensor."""
         super().__init__(device, device_id, event=event)
-        self.data_type = data_type
-        self._unit_of_measurement = DATA_TYPES.get(data_type)
-        self._name = f"{device.type_string} {device.id_string} {data_type}"
-        self._unique_id = "_".join(x for x in (*self._device_id, data_type))
-
-        self._device_class = DEVICE_CLASSES.get(data_type)
-        self._convert_fun = CONVERT_FUNCTIONS.get(data_type, lambda x: x)
+        self.entity_description = entity_description
+        self._name = f"{device.type_string} {device.id_string} {entity_description.key}"
```

Comment:
If you want to go all fancy, you can replace this with `_attr_name` and `_attr_unique_id` and rely on the default properties that return those for name and unique ID.

Reply:
(in a future PR)